src/cryptonote_basic/miner.cpp

```diff
@@ -763,45 +784,54 @@ namespace cryptonote
     return (uint8_t)( ceil( (other * 1.f / total * 1.f) * 100) );    
   }
   //-----------------------------------------------------------------------------------------------------    
-  bool miner::on_battery_power()
+  boost::logic::tribool miner::on_battery_power()
```

Comment:
I made this comment too quickly. The meaning of `indeterminate` is clear in this context.

The only concern would be for future maintenance. tribool overloads `operator&&` and `operator||` which means [short-circuiting is disabled](http://stackoverflow.com/questions/27235555/is-it-compiler-bug-or-my-bug-when-using-boosttribool-in-a-conditional) for left and right arguments when tribool is one such argument.